qiskit/algorithms/time_evolvers/variational/variational_principles/imaginary_mc_lachlan_principle.py

```diff
@@ -0,0 +1,117 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class for an Imaginary McLachlan's Variational Principle."""
+from __future__ import annotations
+
+import warnings
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.primitives import Estimator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .imaginary_variational_principle import ImaginaryVariationalPrinciple
+
+from ....exceptions import AlgorithmError
+from ....gradients import (
+    BaseEstimatorGradient,
+    BaseQGT,
+    DerivativeType,
+    LinCombQGT,
+    LinCombEstimatorGradient,
+)
+
+
+class ImaginaryMcLachlanPrinciple(ImaginaryVariationalPrinciple):
+    """Class for an Imaginary McLachlan's Variational Principle. It aims to minimize the distance
+    between both sides of the Wick-rotated Schrödinger equation with a quantum state given as a
+    parametrized trial state. The principle leads to a system of linear equations handled by a
+    linear solver. The imaginary variant means that we consider imaginary time dynamics.
+    """
+
+    def __init__(
+        self,
+        qgt: BaseQGT | None = None,
+        gradient: BaseEstimatorGradient | None = None,
+    ) -> None:
+        """
+        Args:
+            qfi: Instance of a the GQT class used to compute the QFI.
+                If ``None`` provided, ``LinCombQGT`` is used.
+            gradient: Instance of a class used to compute the state gradient.
+                If ``None`` provided, ``LinCombEstimatorGradient`` is used.
+        """
+
+        self._validate_grad_settings(gradient)
+
+        if gradient is not None and gradient._estimator is not None and qgt is None:
```

Comment:
Both bits here are qgt is None. Is it valid to supply just a qgt. In this case gradient does not seem to get set to any default value if its left at None

Reply:
Updated both inits to decouple the `qgt` from the `gradient` checks.